### ISSUE TITLE
App freezes / turns black screen when switching between apps

### ISSUE THREAD

**Issue by Specy-wot** [GoogleChromeLabs/bubblewrap#273]
**Describe the bug**
I used the website pwabuilder to turn [My webapp ](https://sky-music.herokuapp.com/) Into an apk, It turned out fine but I have issues:
1) the manifest has the orientation set to landscape but in the app it is ignored and the user can rotate the app however they want 
2) When switching from my app to some other app and waiting +-30 seconds, after switching back to my app it is either frozen or has a black screen
3) when changing pages inside the app, sometimes it happens that the app freezes

All those problems can be fixed momentarily by rotating the device, it will unstuck it. 
[Here is my app](https://play.google.com/store/apps/details?id=com.herokuapp.sky_music.twa) 
And here is the manifest 
https://sky-music.herokuapp.com/manifest.json

When using the browser "add to the home" feature the app works as expected and never freezes, rotation is locked on landscape, I filed the report to pwabuilder but they told me to write it here 
**To Reproduce**
1. Download the app I linked 
2. Use it for some seconds then switch to another app (without going home)
3. Use another app for 30/45 seconds 
4. Go back to my app, again without going to the home 

**Expected behavior**
The app should not freeze and work as expected 

**Screenshots**
[Video example ](https://cdn.discordapp.com/attachments/706254334098276402/738681914503987210/20-07-31-10-55-07_576x1024.mp4) 

**Smartphone:**
 - Device: [OnePlus 6]
 - OS: [Android 10]
 - Browser [Chrome]
 - Version [84.0.4147.111]

**Additional context**
Also happens with other user devices who have reported me this issue. 
[This](https://github.com/pwa-builder/PWABuilder/issues/997#event-3617125666)  is the issue I filed on pwabuilder repo

**Comment by andreban** [comment]
Thanks for the detailed bug, @Specy-wot 

> 1. the manifest has the orientation set to landscape but in the app it is ignored and the user can rotate the app however they want

This is not implemented yet, but work is happening on https://github.com/GoogleChrome/android-browser-helper/pull/127. It does depend on another support library being published to stable, so I'd expect it to be a couple of months away or more still.

> 2. When switching from my app to some other app and waiting +-30 seconds, after switching back to my app it is either frozen or has a black screen
> 3. when changing pages inside the app, sometimes it happens that the app freezes

I tried both on a Pixel 3 and Samsung Galaxy A10 with Chrome Stable 84.0.4147.111, but couldn't reproduce. Would mind checking if that's still reproducible when with [Chrome Dev](https://play.google.com/store/apps/details?id=com.chrome.dev) and [Chrome Beta](https://play.google.com/store/apps/details?id=com.chrome.beta)? You'll need to set each as the default browser for testing, to ensure they are handling opening the application.

I also saw this comment on the user reviews, with a user saying they are seeing freezes that also happen in Chrome. Can those be related? Did the user provide more details?
> A bit buggy works well the main concern is that sometimes the game crashes and when I try to open it it freezes and doesn't open. It happens to chrome as well (probably my device)

**Comment by Specy-wot** [comment]
@andreban 
Oh ok I didn't know it wasn't implemented yet, at least I know it's not my fault on that one.

I tried with the two you mentioned and it still happens, I've tested the website often on an actual browser but I don't remember it ever having such issue, the user who sent the review never replied tho so I couldn't investigate.
I always thought it was something on my end but I never managed to figure out what could be causing it. The website doesn't do anything in the background without user input.
